File: app/src/main/java/ru/melodin/fast/api/model/VKLink.java
package ru.melodin.fast.api.model;

import androidx.annotation.NonNull;

import org.json.JSONObject;

import java.io.Serializable;

import ru.melodin.fast.api.model.attachment.VKPhoto;

public class VKLink extends VKModel implements Serializable {

    private static final long serialVersionUID = 1L;

    private String url;
    private String title;
    private String caption;
    private String description;
    private String previewUrl;

    private VKPhoto photo;

    public VKLink(@NonNull JSONObject source) {
        this.url = source.optString("url");
        this.title = source.optString("title");
        this.caption = source.optString("caption");
        this.previewUrl = source.optString("preview_url");
        this.description = source.optString("description");

        JSONObject linkPhoto = source.optJSONObject("photo");
        if (linkPhoto != null) {
            this.photo = new VKPhoto(linkPhoto);
        }
    }

    public String getUrl() {
        return url;
    }

    public String getTitle() {
        return title;
    }

    public String getCaption() {
        return caption;
    }

    public String getDescription() {
        return description;
    }

    public String getPreviewUrl() {
        return previewUrl;
    }

    public VKPhoto getPhoto() {
        return photo;
    }
}

File: app/src/main/java/ru/melodin/fast/api/model/VKLink.kt
package ru.melodin.fast.api.model

import org.json.JSONObject
import ru.melodin.fast.api.model.attachment.VKPhoto
import java.io.Serializable

class VKLink(source: JSONObject) : VKModel(), Serializable {

    val url: String = source.optString("url")
    val title: String = source.optString("title")
    val caption: String = source.optString("caption")
    val description: String = source.optString("description")
    val previewUrl: String = source.optString("preview_url")

    var photo: VKPhoto? = null
        private set

    init {
        val linkPhoto = source.optJSONObject("photo")
        if (linkPhoto != null) {
            this.photo = VKPhoto(linkPhoto)
        }
    }

    companion object {

        private const val serialVersionUID = 1L
    }
}

